lib/endpoint.go

```diff
@@ -139,14 +139,14 @@ func (e *Endpoint) Write(etcd client.KeysAPI) error {
 	return nil
 }
 
-func GetEndpoint(etcd client.KeysAPI, w Workload) (bool, Endpoint, error) {
+func GetEndpoint(etcd client.KeysAPI, w Workload) (*Endpoint, error) {
```

Comment:
Is your long-term plan that the libcalico user will need to pass a KeysAPI on each API call?

Seems like we should be abstracting that from the user of libcalico.